src/eradiate/test_tools/regression.py

```diff
@@ -54,50 +63,172 @@ def regression_test_plots(data: xr.Dataset, filename: PathLike, metric: tuple) -
     plt.close()
 
 
+def reference_converter():
+    """
+    A converter for handling the reference data attribute.
+
+    The produced converter processes a parameter ``value`` as follows:
+
+    * if ``value`` is a string or path like object, it attempts to load
+      a dataset from that location and returns ``None`` if that fails.
+    * if ``value`` is a xarray dataset, it is returned directly.
+
+    Returns
+    -------
+    callable
+        Generated converter.
+    """
+
+    def f(value):
+        if isinstance(value, (str, os.PathLike)):
+            if not os.path.isfile(value):
+                return None
+            return xr.load_dataset(value)
+        elif isinstance(value, xr.Dataset):
+            return value
+
+    return f
+
+
 @attr.s
-class RenderTest(ABC):
+class RegressionTest(ABC):
     """
     Common interface for tests based on the comparison of a result array against
     reference values.
     """
 
-    value: xr.DataArray = attr.ib()
-    reference: xr.DataArray = attr.ib()
-    threshold: float = attr.ib(default=0.05)
-    archive_filename: PathLike = attr.ib(default=None)
+    value: xr.Dataset = attr.ib(validator=attr.validators.instance_of(xr.Dataset))
+    reference: xr.Dataset = attr.ib(
+        default=None,
+        converter=attr.converters.optional(reference_converter()),
```

Comment:
With the aforementioned changes:
```suggestion
        converter=attr.converters.optional(reference_converter),
```